### DISCUSSION TITLE
How to get network Id as soon as user gets connected via walletConnect, using wagmi

### DISCUSSION THREAD

**Discussion by puneettiwariofficial** [WalletConnect/walletconnect-monorepo#2836] [Developer Support] [answered]
Here is my code
```
const { address, isConnected } = useAccount();
const { chain, chains } = useNetwork();

useEffect(() => {
    console.log(chain, "imchain", chains, isConnected);
    if (!details?.walletKey && address && isConnected && chain) {
      localStorage.setItem("loginProvider", "walletConnect");
      setAccounts([address]);
      findIfUserIsRegistered(address);
    }
  }, [chain, details]);
  ```

After when Web3Modal modal appears and user is logged in, it logs in the following manner.
The user is actually on Chain ID 137, but it shows 80001 initially, though the user is already connected.

Is there anything I am doing wrong?

Please go through the screenshot
<img width="760" alt="Screenshot 2023-06-28 at 5 13 13 PM" src="https://github.com/wagmi-dev/wagmi/assets/104764504/9a8364e4-84ae-49c5-8475-a243dc7edc58">

**Comment by boidushya** [comment]
Hey, what chains did you pass through wagmi's WalletConnect connector?
Also could you provide a code snippet for initialising the WalletConnectConnector?

**Reply by puneettiwariofficial** [reply]
sure here it is,
I passed PolygonMumbai (80001) chain while initialising. 
```
import {
  EthereumClient,
  w3mConnectors,
  w3mProvider,
} from "@web3modal/ethereum";
import { Web3Modal } from "@web3modal/react";
import { configureChains, createConfig, WagmiConfig } from "wagmi";
import { polygon, polygonMumbai } from "wagmi/chains";
import { WALLET_CONNECT_PROJECT_ID } from "./config";
const chains = [polygonMumbai];

const { publicClient } = configureChains(chains, [
  w3mProvider({ projectId: WALLET_CONNECT_PROJECT_ID }),
]);
const wagmiConfig = createConfig({
  autoConnect: false,
  connectors: w3mConnectors({
    projectId: WALLET_CONNECT_PROJECT_ID,
    chains,
  }),
  publicClient,
});
const ethereumClient = new EthereumClient(wagmiConfig, chains);
```

**Reply by boidushya** [reply]
Taking a look, trying to reproduce this

**Comment by xzilja** [comment, marked as answer]
Unfortunately this is a known limitation in MetaMask, they chose to implement everything in a proprietary manner where chain switch happens only after initial connection (to chain in MetaMask's wallet first)
